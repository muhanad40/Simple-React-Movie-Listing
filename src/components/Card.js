import React, { Component } from 'react';

const renderGenres = (genreIds, genres) => {
  return genreIds.map(genreId => {
      return genres.filter(genre => genre.id === genreId)[0]
    })
    .map(genre => genre.name)
    .join(', ')
};

const Card = (props) => (
  <a href="#" className="card">
    <div className="card__image" style={{ backgroundImage: `url(${ props.poster_path })` }}>
      <span className="card__year">{ props.release_date.split('-')[0] }</span>
    </div>

    <div className="card__info">
      <div className="card__copy">
        <span className="card__title">{ props.title }</span>

        <p className="card__genres">{ renderGenres(props.genre_ids, props.genres) }</p>
      </div>

      <div className="card__rating">
        { props.vote_average }
      </div>
    </div>
  </a>
);

export default Card;
