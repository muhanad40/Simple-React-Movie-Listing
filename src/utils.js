export function getGenre(genreId, genres) {
  return genres.filter(genre => genre.id === genreId)[0];
}

export function getGenres(genreIds, genres) {
  return genreIds.map(genreId => {
      return getGenre(genreId, genres);
    })
}
