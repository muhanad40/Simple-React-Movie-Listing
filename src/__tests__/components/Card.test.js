import Card from '../../components/Card';
import mockNowPlayingData from '../../__mocks__/now_playing.json';
import mockGenres from '../../__mocks__/genres.json';
import { setupCreator } from '../utils';

const movieData = mockNowPlayingData.results[0];
const setup = setupCreator(Card, {
  ...movieData,
  genres: mockGenres.genres
});

describe('Card component', () => {
  let instance;
  let wrapper

  beforeAll(() => {
    ({ instance, wrapper } = setup());
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});