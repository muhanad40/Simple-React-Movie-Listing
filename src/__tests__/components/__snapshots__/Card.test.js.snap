// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    adult={false}
    backdrop_path="/ndlQ2Cuc3cjTL7lTynw6I4boP4S.jpg"
    genre_ids={
      Array [
        28,
        30,
      ]
    }
    genres={
      Array [
        Object {
          "id": 28,
          "name": "Action",
        },
        Object {
          "id": 29,
          "name": "Drama",
        },
        Object {
          "id": 30,
          "name": "Adventure",
        },
      ]
    }
    id={297761}
    original_language="en"
    original_title="Suicide Squad"
    overview="From DC Comics comes the Suicide Squad, an antihero team of incarcerated supervillains who act as deniable assets for the United States government, undertaking high-risk black ops missions in exchange for commuted prison sentences."
    popularity={48.261451}
    poster_path="/e1mjopzAS2KNsvpbpahQ1a6SkSn.jpg"
    release_date="2016-08-03"
    title="Suicide Squad"
    video={false}
    vote_average={5.91}
    vote_count={1466}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="card__image"
          style={
            Object {
              "backgroundImage": "url(/e1mjopzAS2KNsvpbpahQ1a6SkSn.jpg)",
            }
          }
        >
          <span
            className="card__year"
          >
            2016
          </span>
        </div>,
        <div
          className="card__info"
        >
          <div
            className="card__copy"
          >
            <span
              className="card__title"
            >
              Suicide Squad
            </span>
            <p
              className="card__genres"
            >
              Action, Adventure
            </p>
          </div>
          <div
            className="card__rating"
          >
            5.91
          </div>
        </div>,
      ],
      "className": "card",
      "href": "#",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <span
            className="card__year"
          >
            2016
          </span>,
          "className": "card__image",
          "style": Object {
            "backgroundImage": "url(/e1mjopzAS2KNsvpbpahQ1a6SkSn.jpg)",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "2016",
            "className": "card__year",
          },
          "ref": null,
          "rendered": "2016",
          "type": "span",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="card__copy"
            >
              <span
                className="card__title"
              >
                Suicide Squad
              </span>
              <p
                className="card__genres"
              >
                Action, Adventure
              </p>
            </div>,
            <div
              className="card__rating"
            >
              5.91
            </div>,
          ],
          "className": "card__info",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  className="card__title"
                >
                  Suicide Squad
                </span>,
                <p
                  className="card__genres"
                >
                  Action, Adventure
                </p>,
              ],
              "className": "card__copy",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Suicide Squad",
                  "className": "card__title",
                },
                "ref": null,
                "rendered": "Suicide Squad",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Action, Adventure",
                  "className": "card__genres",
                },
                "ref": null,
                "rendered": "Action, Adventure",
                "type": "p",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 5.91,
              "className": "card__rating",
            },
            "ref": null,
            "rendered": 5.91,
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "a",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="card__image"
            style={
              Object {
                "backgroundImage": "url(/e1mjopzAS2KNsvpbpahQ1a6SkSn.jpg)",
              }
            }
          >
            <span
              className="card__year"
            >
              2016
            </span>
          </div>,
          <div
            className="card__info"
          >
            <div
              className="card__copy"
            >
              <span
                className="card__title"
              >
                Suicide Squad
              </span>
              <p
                className="card__genres"
              >
                Action, Adventure
              </p>
            </div>
            <div
              className="card__rating"
            >
              5.91
            </div>
          </div>,
        ],
        "className": "card",
        "href": "#",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <span
              className="card__year"
            >
              2016
            </span>,
            "className": "card__image",
            "style": Object {
              "backgroundImage": "url(/e1mjopzAS2KNsvpbpahQ1a6SkSn.jpg)",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "2016",
              "className": "card__year",
            },
            "ref": null,
            "rendered": "2016",
            "type": "span",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="card__copy"
              >
                <span
                  className="card__title"
                >
                  Suicide Squad
                </span>
                <p
                  className="card__genres"
                >
                  Action, Adventure
                </p>
              </div>,
              <div
                className="card__rating"
              >
                5.91
              </div>,
            ],
            "className": "card__info",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span
                    className="card__title"
                  >
                    Suicide Squad
                  </span>,
                  <p
                    className="card__genres"
                  >
                    Action, Adventure
                  </p>,
                ],
                "className": "card__copy",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Suicide Squad",
                    "className": "card__title",
                  },
                  "ref": null,
                  "rendered": "Suicide Squad",
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Action, Adventure",
                    "className": "card__genres",
                  },
                  "ref": null,
                  "rendered": "Action, Adventure",
                  "type": "p",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 5.91,
                "className": "card__rating",
              },
              "ref": null,
              "rendered": 5.91,
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "a",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
